<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#find-mislabels" data-toc-modified-id="find-mislabels-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>find mislabels</a></span></li><li><span><a href="#grab-bad-trials" data-toc-modified-id="grab-bad-trials-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>grab bad trials</a></span></li><li><span><a href="#run-update-script" data-toc-modified-id="run-update-script-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>run update script</a></span></li><li><span><a href="#changed-trials" data-toc-modified-id="changed-trials-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>changed trials</a></span></li></ul></div>

In [137]:
exptGrps,roi_filters = creb.load_creb_experiments(filter_type=None) 

LOADED EXPERIMENT GROUPS: CREB,dn-CREB,Control
Using filter...None


In [138]:
days=[]
mouse=[]
sessions=[]
conditions=[]
tis=[]
bp=[]

for e in exptGrps[0]:
    days.append(e.get('day'))
    mouse.append(e.parent.get('mouseID'))
    sessions.append(e.get('session'))
    conditions.append(e.get('condition'))
    tis.append(e.get('trial_id'))
    bp.append(e.find('trial').behaviorDataPath())
    
    
df = pd.DataFrame(zip(mouse,conditions,days,sessions,tis,bp), 
                  columns=['mouseID','condition','day','session','trialID','bp'])

# add condition day
df["condition_day"] = df["condition"] + "_" + df["day"]

# find mislabels

In [139]:

df.groupby(["condition_day","mouseID"]).count()

condition  day  session  trialID  bp
condition_day mouseID                                      
1_1           mjd500           4    4        4        4   4
              mjd503           3    3        3        3   3
              mjd511           3    3        3        3   3
              mjd535           3    3        3        3   3
              mjd537           3    3        3        3   3
1_2           mjd500           2    2        2        2   2
              mjd503           3    3        3        3   3
              mjd511           3    3        3        3   3
              mjd535           3    3        3        3   3
              mjd537           3    3        3        3   3
1_3           mjd500           3    3        3        3   3
              mjd503           2    2        2        2   2
              mjd511           3    3        3        3   3
              mjd535           3    3        3        3   3
              mjd537           3    3        3        3   3
2_4           mjd500           3    3        3        3   3
              mjd503           3    3        3        3   3
              mjd511           3    3        3        3   3
              mjd535           3    3        3        3   3
              mjd537           3    3        3        3   3
2_5           mjd500           3    3        3        3   3
              mjd503           2    2        2        2   2
              mjd511           3    3        3        3   3
              mjd535           3    3        3        3   3
              mjd537           3    3        3        3   3
2_6           mjd500           3    3        3        3   3
              mjd503           3    3        3        3   3
              mjd511           3    3        3        3   3
              mjd535           3    3        3        3   3
              mjd537           3    3        3        3   3

# grab bad trials

In [134]:
to_change = df.loc[(df["condition"]=='1') & (df["day"]=='4')] # & (df["mouseID"]=='mjd535')]
to_change

mouseID condition day session trialID  \
79  mjd535         1   4       1   13207   
80  mjd535         1   4       2   13206   
81  mjd535         1   4       3   13205   

                                                   bp condition_day  
79  /home/matt/creb/behavior/data/mjd535/mjd535_20...           1_4  
80  /home/matt/creb/behavior/data/mjd535/mjd535_20...           1_4  
81  /home/matt/creb/behavior/data/mjd535/mjd535_20...           1_4

In [135]:
to_change.bp.values

array(['/home/matt/creb/behavior/data/mjd535/mjd535_20180621113148.pkl',
       '/home/matt/creb/behavior/data/mjd535/mjd535_20180621130639.pkl',
       '/home/matt/creb/behavior/data/mjd535/mjd535_20180621141259.pkl'],
      dtype=object)

# run update script

In [136]:
! cd /home/matt/code/analysis/automaticScripts
for i,r in to_change.iterrows():
    print(r.trialID)
    ! cd /home/matt/code/analysis/automaticScripts to_change&& \
      python processJsonBehaviorData.py -f {r.bp.replace('pkl','tdml')} -t {r.trialID} -o

13207
/home/matt/anaconda2/lib/python2.7/site-packages/seaborn/apionly.py:9: UserWarning: As seaborn no longer sets a default style on import, the seaborn.apionly module is deprecated. It will be removed in a future version.
  warnings.warn(msg, UserWarning)
/home/matt/anaconda2/lib/python2.7/site-packages/dask/config.py:168: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  data = yaml.load(f.read()) or {}
/home/matt/code/analysis/lab/misc/lfp_helpers.py:16: UserWarning: rhd package not found, lab.misc.lfp_helpers.py will not work
  warnings.warn("rhd package not found, lab.misc.lfp_helpers.py will not work")
pickling /home/matt/creb/behavior/data/mjd535/mjd535_20180621113148.tdml
loading trial 13207: /home/matt/creb/behavior/data/mjd535/mjd535_20180621113148.tdml
13206
/home/matt/anaconda2/lib/python2.7/site-packages/seaborn/apionly.py:9: UserWarning: As seaborn no longer

# changed trials

61	mjd537	1	4	1	13209	/home/matt/creb/behavior/data/mjd537/mjd537_20...
62	mjd537	1	4	2	13211	/home/matt/creb/behavior/data/mjd537/mjd537_20...
63	mjd537	1	4	3	13210	/home/matt/creb/behavior/data/mjd537/mjd537_20...
79	mjd535	1	4	1	13207	/home/matt/creb/behavior/data/mjd535/mjd535_20...
80	mjd535	1	4	2	13206	/home/matt/creb/behavior/data/mjd535/mjd535_20...
81	mjd535	1	4	3	13205	/home/matt/creb/behavior/data/mjd535/mjd535_20...

In [21]:
day_paired = [grp.pair('same group', groupby=['X_session']).pair(
        'consecutive groups', groupby=['X_condition', 'X_day']) for grp in
        exptGrps]

In [48]:
len(day_paired[0])

83

In [43]:
for e in day_paired[0]:
    print(e.get('tSeriesDirectory'))

/home/matt/creb/CREB3-4/mjd511/20171209-1-1-1
/home/matt/creb/CREB3-4/mjd511/20171209-1-2-1
/home/matt/creb/CREB3-4/mjd511/20171209-1-3-1
/home/matt/creb/CREB3-4/mjd511/20171210-2-1-1
/home/matt/creb/CREB3-4/mjd511/20171210-2-2-1
/home/matt/creb/CREB3-4/mjd511/20171210-2-3-1
/home/matt/creb/CREB3-4/mjd511/20171211-3-1-1
/home/matt/creb/CREB3-4/mjd511/20171211-3-2-1
/home/matt/creb/CREB3-4/mjd511/20171211-3-3-1
/home/matt/creb/CREB3-4/mjd511/20171212-4-1-2
/home/matt/creb/CREB3-4/mjd511/20171212-4-2-2
/home/matt/creb/CREB3-4/mjd511/20171212-4-3-2
/home/matt/creb/CREB3-4/mjd511/20171213-5-1-2
/home/matt/creb/CREB3-4/mjd511/20171213-5-2-2
/home/matt/creb/CREB3-4/mjd511/20171213-5-3-2
/home/matt/creb/CREB3-4/mjd511/20171214-6-1-2
/home/matt/creb/CREB3-4/mjd511/20171214-6-2-2
/home/matt/creb/CREB3-4/mjd511/20171214-6-3-2
/home/matt/creb/CREB3-4/mjd503/20171129-HR-1-1-1
/home/matt/creb/CREB3-4/mjd503/20171129-HR-1-2-1
/home/matt/creb/CREB3-4/mjd503/20171129-HR-1-3-1
/home/matt/creb/CREB3-4/m

In [28]:
len(day_paired[2])

74